src/components/Idler.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+
+import { PureComponent } from 'react'
+import { createStructuredSelector } from 'reselect'
+import { connect } from 'react-redux'
+import { hasPasswordSelector, autoLockTimeoutSelector } from 'reducers/settings'
+import debounce from 'lodash/debounce'
+import { lock } from 'reducers/application'
+
+type Props = {
+  autoLockTimeout: number,
+  hasPassword: boolean,
+  lock: Function,
+}
+
+type State = {
+  lastAction: number,
+}
+
+const mapStateToProps = createStructuredSelector({
+  autoLockTimeout: autoLockTimeoutSelector,
+  hasPassword: hasPasswordSelector,
+})
+
+const mapDispatchToProps = {
+  lock,
+}
+
+class Idler extends PureComponent<Props, State> {
+  state = {
+    lastAction: -1,
+  }
+
+  componentDidMount() {
+    window.addEventListener('keydown', this.debounceOnChange)
+    window.addEventListener('mouseover', this.debounceOnChange)
+    this.interval = setInterval(this.checkForAutoLock, 10000)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('keydown', this.debounceOnChange)
+    window.removeEventListener('mouseover', this.debounceOnChange)
+    clearInterval(this.interval)
+  }
+
+  interval: IntervalID
+
+  debounceOnChange = debounce(_ => this.idleTimeHandler(), 1000)
```

Comment:
this debounce is not "free" at unmount time. not sure if we have such unmount case, but for the beauty of it, you could `this.debounceOnChange.cancel()` at unmount to be sure the inner function won't get called after.

Reply:
Added it to the current pr